File: Counting-element/counting-elements.rb

def counting_elements(arr)
    count = 0
    new_arr = arr.sort
    (0..new_arr.length - 1).each do |i|
        (i + 1..new_arr.legth -1).each do |j|
            if new_arr[j] == new_arr[i] + 1
                count = count +1
            end
        end
    end
    return count
end

puts counting_elements([2, 8, 5, 10, 9, 6])

File: Counting-element/counting-elements.js
var countingElements = function (array) {
  document.getElementById("array").innerHTML = "[" + array + "]";
  let arraySorted = array.sort();
  let count = 0;
  for (let i = 0; i < array.length; i++) {
    for (let j = i + 1; j < arraySorted.length; j++) {
      if (arraySorted[j] == arraySorted[i] + 1) {
        count++;
      }
    }
  }
  document.getElementById("answer").innerHTML = count;
};
